src/providers/mbed_provider/mod.rs

```diff
@@ -70,9 +68,8 @@ impl MbedProvider {
         }
         let mbed_provider = MbedProvider {
             key_info_store,
-            local_ids: RwLock::new(HashSet::new()),
             key_handle_mutex: Mutex::new(()),
-            key_slot_semaphore: Semaphore::new(PSA_KEY_SLOT_COUNT),
+            id_counter: AtomicU32::new(key::PSA_KEY_ID_USER_MIN),
```

Comment:
This is wrong - the value of the counter has to be persisted. Which makes me wonder how we'll manage that........

Reply:
Oh, we can implement `Drop` on the `MbedProvider` and write it somewhere - sounds like we need some new "service" for persisting arbitrary data. Then we can read it back when the service is started

Reply:
Finding the highest currently used ID is taken care of (lines 100 to 102). While looping through all keys in the store to check they are still valid, it keeps track of the highest ID in use.

Reply:
Ah! makes sense, didn't make the correlation between the two!